### ISSUE TITLE
Failed to build Node.js application.

### ISSUE THREAD

**Issue by monaka** [deis/builder#371]
Yesterday I installed Deis Workflow on Azure VM / CoreOS / Hyperkube.

And I'm trying to deploy static web provider (https://github.com/nulltask/heroku-static-provider.git) and failed. The build log is below.
It may failed to [resolve the version of Node.js](https://github.com/heroku/heroku-buildpack-nodejs/blob/v90/lib/binaries.sh#L16). 

There runs an network application (but no DNS dependent) deployed by 'deis pull` on the same node.
It is working well. So the node keep network connections between pods and the Internet.

My first impression, I guess this issue is caused by DNS settings.
Could anyone give me some hints?

```
-----> Node.js app detected
       
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=true
       NODE_ENV=production
       NODE_MODULES_CACHE=true
       
-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)
       
       Resolving node version (latest stable) via semver.io...
       Downloading and installing node ...
       Unable to download node ; does it exist?
       
-----> Build failed
       
       We're sorry this build is failing! You can troubleshoot common issues here:
       https://devcenter.heroku.com/articles/troubleshooting-node-deploys
       
       Some possible problems:
       
       - Node version not specified in package.json
       https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
       
       Love,
       Heroku
```

**Comment by monaka** [comment]
I got it. Just my mistake.
My builder works powerful after fixing '--cluster-dns=` parameter for Kubelet.
I close this issue as it is not caused by Builder.

**Comment by reutopiaer** [comment]
@monaka I got the same problem :(
Can your tell me how to fix it?

**Comment by monaka** [comment]
@reutopiaer , my case is...

My k8s cluster is built on Hyperkube. So I must set `--cluster-dns=` as the IP address to SkyDNS.
But I set an wrong IP address there.

In addition, I got an another rare case.
SkyDNS on my cluster was down. After deleting (and auto restarting) related pods, it worked well again.

Anyway,  try to run `dig github.com @{SkyDNS's IP address}'. Not the public DNS but SkyDNS in your k8s cluster.
It will be an entry point to fix your issue.
